### ISSUE TITLE
require-context-role considers named block as parent element

### ISSUE THREAD

**Issue by kangax** [ember-template-lint/ember-template-lint#2462]
`require-context-role` rule seems to consider a named block as one of the ancestor elements. Since the rule asks for an immediate parent to have a certain role, it reports a false positive in cases when there's a named block in its parent chain.

Given (roughly) this markup:

```hbs
<typeahead.list role="list">
  <:content as |items|>
    {{#each items as |job idx|}}
      <result role="listitem">
        ...
      </result>
    {{/each}}
    </:content>
</typeahead.list>
```

The error is:

>  135:22  error  You have an element with the role of "listitem" but it is missing the required (immediate) parent element of "[group, list]". Reference: https://www.w3.org/TR/wai-aria-1.0/roles#listitem.  require-context-role

Removing named block makes the error disappear.

(tested with ember-template-lint 3.12.0)

**Comment by lifeart** [comment]
@kangax thank you for report!
